ls1intum/Athene
# Feature/merge clusters

Changes to the data that is exported in ClusteringResource.py. In addition to the Clusters, following will also be exported:
- Distance matrix for the distances between all text blocks
- Cluster tree representing the hierarchy of clusters

### Testing:
- I added an additional test case to test_clustering.py
- [This](https://github.com/ls1intum/Artemis/pull/1934) PR in Artemis has the complementary branch (Build currently fails in github because of dependancy to this branch, tests should work locally)
    - TextSimilarityClusteringTest tests that the new entities are correctly received in Artemis
    - TextClusteringTest test that they are correctly saved to the database

**Manual Testing:**
1) Start Athene using docker-compose
2) Run Artemis Server and Client ([This](https://github.com/ls1intum/Artemis/pull/1934) branch)
3) Create a text exercise with automatic feedback suggestions enabled and due date a couple of minutes later
4) Submit a text with > 10 sentences (a couple of paragraphs copied from wikipedia etc.)
5) Wait for the due date of the exercise to pass so that clustering can be performed.
6) Check the database tables: text_block, text_cluster, text_pairwise_distance, text_tree_node should have entries

**There is no new feature in this branch, it should only be checked that these entities are correctly stored.**
